src/libsyntax/parse/token.rs

```diff
@@ -324,6 +329,16 @@ impl Token {
         }
     }
 
+    /// Returns `true` if the token is any literal or the boolean idents
+    pub fn is_lit_or_bool(&self) -> bool {
+        match *self {
+            Literal(..) => true,
+            Ident(ident, false) if ident.name == "true" => true,
+            Ident(ident, false) if ident.name == "false" => true,
```

Comment:
Nit: `keywords::True.name()`/`keywords::False.name()` instead of `"true"`/`"false"`.